js/core/dom_component_with_template.js

```diff
@@ -1,290 +1,76 @@
         }
     },
 
-    _getAnonymousTemplateName: function() {
-        return ANONYMOUS_TEMPLATE_NAME;
-    },
-
-    _createTemplateIfNeeded: function(templateSource) {
-        var templateKey = function(templateSource) {
-            return (isRenderer(templateSource) && templateSource[0]) || templateSource;
-        };
-
-        var cachedTemplate = this._tempTemplates.filter(function(t) {
-            templateSource = templateKey(templateSource);
-            return t.source === templateSource;
-        })[0];
-        if(cachedTemplate) return cachedTemplate.template;
-
-        var template = this._createTemplate(templateSource);
-        this._tempTemplates.push({ template: template, source: templateKey(templateSource) });
-        return template;
-    },
-
-    _createTemplate: function(templateSource) {
-        templateSource = typeof templateSource === "string" ? normalizeTemplateElement(templateSource) : templateSource;
-        return this.option("integrationOptions.createTemplate")(templateSource);
-    },
-
     _getTemplateByOption: function(optionName) {
         return this._getTemplate(this.option(optionName));
     },
 
     _getTemplate: function(templateSource) {
-        if(isFunction(templateSource)) {
-            return new FunctionTemplate(function(options) {
-                var templateSourceResult = templateSource.apply(this, this._getNormalizedTemplateArgs(options));
-
-                if(!isDefined(templateSourceResult)) {
-                    return new EmptyTemplate();
-                }
-
-                var dispose = false;
-                var template = this._acquireTemplate(templateSourceResult, function(templateSource) {
-                    if(templateSource.nodeType || isRenderer(templateSource) && !$(templateSource).is("script")) {
-                        return new FunctionTemplate(function() {
-                            return templateSource;
-                        });
-                    }
-                    dispose = true;
-                    return this._createTemplate(templateSource);
-                }.bind(this));
-
-                var result = template.render(options);
-                dispose && template.dispose && template.dispose();
-                return result;
-            }.bind(this));
-        }
-
-        return this._acquireTemplate(templateSource, this._createTemplateIfNeeded.bind(this));
+        const that = this;
+        const templates = this.option('integrationOptions.templates');
+        const isAsyncTemplate = this.option('templatesRenderAsynchronously');
+        const skipTemplates = this.option('integrationOptions.skipTemplates');
+
+        return this._templateManager.getTemplate(
+            templateSource,
+            templates,
+            {
+                isAsyncTemplate,
+                skipTemplates
+            },
+            that
```

Comment:
I guess that can be replaced with `this`, as there is no closure and the argument is passed directly